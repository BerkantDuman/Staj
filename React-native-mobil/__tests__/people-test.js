/**
 * @format
 */

import 'react-native';
import React from 'react';
import Register from '../src/screens/register';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import People from '../src/screens/people';
import Data from '../src/screens/home';
//import 'isomorphic-fetch';

test('Api test case', async function (){
    global.fetch = jest.fn().mockImplementation(()=>{
        var p = new Promise((resolve, reject)=> {
            resolve({
                json: function() {
                    return {id :2}
                }
            })
        })
        return p;
    })
    const response = await Data.all();
    console.warn(response);
    expect(response.id).toBe(2);
})


test('renders correctly', () => {
    const tree = renderer.create(<People />).toJSON();
    expect(tree).toMatchSnapshot();
  });
