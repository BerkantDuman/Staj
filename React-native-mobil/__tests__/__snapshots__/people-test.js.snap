// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "pink",
      "height": "100%",
      "justifyContent": "center",
    }
  }
>
  <RCTScrollView>
    <View>
      <Text
        style={
          Object {
            "color": "#424242",
            "fontSize": 24,
            "marginLeft": 10,
            "marginTop": 15,
          }
        }
      >
        Search by ID:
      </Text>
      <TextInput
        allowFontScaling={true}
        onChangeText={[Function]}
        placeholder="Find person with ID"
        rejectResponderTermination={true}
        style={
          Object {
            "backgroundColor": "white",
            "borderRadius": 8,
            "color": "black",
            "marginHorizontal": 10,
            "marginVertical": 15,
          }
        }
        underlineColorAndroid="transparent"
        value={null}
      />
      <View
        style={
          Object {
            "borderRadius": 8,
            "marginHorizontal": 10,
            "marginVertical": 15,
          }
        }
      >
        <View
          accessibilityRole="button"
          accessibilityStates={Array []}
          accessible={true}
          clickable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Array [
                Object {},
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#007AFF",
                    "fontSize": 18,
                    "padding": 8,
                    "textAlign": "center",
                  },
                ]
              }
            >
              Search
            </Text>
          </View>
        </View>
      </View>
      <RCTScrollView>
        <View />
      </RCTScrollView>
    </View>
  </RCTScrollView>
</View>
`;
